### ISSUE TITLE
Backslash in columns.data is matched as a period.

### ISSUE THREAD

**Issue by garynthompson** [DataTables/DataTables#869]
Hi,

I haven't actually done a pull request on github before, but if you prefer I do that instead then please let me know and I will learn.  I'm suggesting below such a tiny change, I'm not sure if it would even be considered a contribution.

At the location below is a function which does the replacing of "\\." to "." as per the columns.data manual.

https://github.com/DataTables/DataTables/blob/master/media/js/jquery.dataTables.js#L2541

The replace string is as follows:

return s.replace(/\\./g, '.');

What I believe is an error as the double backslash is escaping the backslash so looking for a literal "\" which is then followed by a period which in regex matches any single character.

So the results is that the string "Hello\\There\\." gets turned into "Hello.here.".

What I believe this line should be is:

return s.replace(/\\\./g, '.');

The third backslash now escapes the '.' 

Pure JS example:
https://jsfiddle.net/tdk0y9mb/2/

Cheers,

Gary

**Comment by DataTables** [comment]
Hi Gary,

Thanks for bringing this up - good catch. That regex does look incorrect. Let me look into it and I'll get back to you.

Allan

**Comment by garynthompson** [comment]
Sorry, there's an error in my original post.  The jsfiddle shows the before / after.

Sorry, I didn't escape my comments in github.

Before:

`
return s.replace(/\\./g, '.');
`

After:

`
return s.replace(/\\\./g, '.');
`

**Comment by DataTables** [comment]
I've committed a fix and test to catch this error (should I ever screw up with it again in future). Thanks again for flagging this up - much appreciated!

Purely because I'm nosey - how on earth did you spot that? Do you have an object that uses `\` in the property names?

**Comment by garynthompson** [comment]
Yeah, pretty much. I'm linking to a third party database that uses a
Windows style path name for referencing data points.  I thought about
encoding slashes as I know they will pay havoc somewhere, but haven't had
time to find a way to support whatever I was going to use to encode. It's a
bit of a nuisance as they allow pretty much every character under the sun
in their strings.

The table was blank and I thought initially I had the columns and data
definition wrong.  I'm using a column render call back so I put a break
point in there, found data was none going into the render function the
column data source also started with a period and had slashes.  I looked
online and find I didn't read the manual for data source, put in the
required slashes in for the period and it still didn't work.

So I stepped up the call stack and followed the processing to find out what
lookup was being performed (in debug) and found the conversion was adding
extra period in the data source string.  Sadly, I was too lazy to get the
non minified version, I used chrome debug pretty view of min js.  I think
this actually helped because I want distracted looking and trying to
interpret the code, I was just following variables and their values.

Um, long story long.

Thanks for your efforts on data tables.
